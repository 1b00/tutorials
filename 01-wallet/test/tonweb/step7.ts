import dotenv from "dotenv";
dotenv.config({ path: "../../../.env" });

import { mnemonicToKeyPair } from "tonweb-mnemonic";
import TonWeb from "tonweb";

async function step7() {
  const mnemonic = process.env.MNEMONIC; // your 24 secret words (replace ... with the rest of the words)
  const key = await mnemonicToKeyPair(mnemonic!.split(" "));

  const tonweb = new TonWeb();
  const WalletClass = tonweb.wallet.all["v4R2"]; // notice the correct wallet version here
  const wallet = new WalletClass(undefined!, {
    publicKey: key.publicKey
  });

  const walletAddress = await wallet.getAddress();
  console.log(walletAddress.toString(true, true, true));
}

step7();