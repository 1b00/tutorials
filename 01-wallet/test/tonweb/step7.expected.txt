EQCJRglfvsQzAIF0UAhkYH6zkdGPFxVNYMH1nPTN_UpDqPrA
workchain: 0
